kube-derive/src/cel_schema.rs

```diff
@@ -189,11 +205,12 @@ mod tests {
     #[test]
     fn test_derive_validated_full() {
         let input = quote! {
-            #[derive(CELSchema)]
+            #[derive(KubeSchema)]
             #[cel_validate(rule = "true".into())]
             struct FooSpec {
                 #[cel_validate(rule = "true".into())]
-                foo: String
+                #[merge_strategy(kind = ListMerge::Atomic)]
```

Comment:
while this is a nice interface, it strikes me as a proc macro with many magic properties to match on that might not be easy to discover.

we could maybe factor it out under an `x-kube` umbrella entry point key to serve as an indicator to users about what does x-kubernetes properties in general, and it would be nice for us to document in a block.

e.g. maybe something like

```rs
struct Foo {
    #[x-kube-validate(rule = "true".into())]
    bar: String
    #[x-kube-merge(kind = ListMerge::Atomic)]
    baz: Vec<String>
}
```

(there are also a couple of other x-kubernetes stuff we don't do yet like [int-or-string](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#intorstring) so maybe that also could go under there)

Reply:
Is it ok to have it all under single attribute?
```rust
#[x_kube(rule = “true”.into(), merge_strategy = ListMerge::Atomic)]
```

While `kube` still may be used, `x-kube` can be used and shared between future extensions.

Reply:
hm, that might work, yeah.

i was initially a bit scared that this might make it harder to put "two argument" things under there, because each key (rule, merge_strategy, etc) is implied to have one value. but you've already found solutions for 2 things like that (rule taking a smart struct that can embed a `message`, and merge strat embeds both the list-type and list-map-keys via smart enums)